app/src/main/java/com/github/polypoly/app/base/game/rules_and_lobby/GameRules.kt

```diff
@@ -0,0 +1,37 @@
+package com.github.polypoly.app.base.game.rules_and_lobby
+
+import com.github.polypoly.app.base.game.location.Zone
+import com.github.polypoly.app.map.LocationRepository
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.hours
+
+/**
+ * A class that represent the rules of a [Game]
+ * @property gameMode The game mode of the future game
+ * @property minimumNumberOfPlayers The minimum number of players needed to start the game
+ * @property maximumNumberOfPlayers The maximum number of players that can join the lobby
+ * @property roundDuration The duration of a round in seconds
+ * @property maxRound The maximum number of round before the game end. This settings is available
+ * only in RICHEST_PLAYER mode. Is null if an other game mode is selected.
+ * @property gameMap The map of the game whit the different zones available
+ * @property initialPlayerBalance The initial balance of money of the players
+ */
+data class GameRules (
+    val gameMode: GameMode = GameMode.RICHEST_PLAYER,
+    val minimumNumberOfPlayers: Int = 3,
+    val maximumNumberOfPlayers: Int = 7,
+    val roundDuration: Int = 0,
+    val maxRound: Int? = null,
```

Comment:
I don't agree, for me if the nullable variables make sense to be nullable we should keep them. Not setting this variable to null when you don't need to use it is taking the risk of using it unintentionally and no warning is indicated. What we must limit as much as possible are the `!!` which are a bad practice, the code must be a maximum null safety. Besides there is a list of good practices that allow to avoid `!!` by making the code null safe like for example assigning the nullable attribute to a local variable at the start of the function and test if the local variable is null. Normally in this case the code compiles without `!!`.

Reply:
This is the page for Kotlin on null safety: https://kotlinlang.org/docs/null-safety.html